### PR TITLE
Support auth for custom users and roles in eck-elasticsearch Helm chart

### PR DESCRIPTION
**Pull request by oziey88**
Adding "auth" block to Elasticsearch template to use a secret and manage users and roles

### CODE REVIEW

**Comment by oziey88** [comment]
Already signed the agreement.

**Comment by thbkrkr** [comment]
Jenkins test this please

**Comment by thbkrkr** [comment]
Buildkite test this please

File: deploy/eck-elasticsearch/values.yaml [hunk 1]
```diff
@@ -28,6 +28,10 @@ labels: {}
 #
 annotations: {}
 
+# Control the Elasticsearch Users and Roles
+# ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-users-and-roles.html
```
Comment (thbkrkr) [inline deploy/eck-elasticsearch/values.yaml:32]:
```suggestion
# Settings for configuring Elasticsearch users and roles.
# ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-users-and-roles.html
#
```

**Review by thbkrkr — CHANGES_REQUESTED** [review]
Could you please add a unit test?

```diff
diff --git a/deploy/eck-elasticsearch/templates/tests/elasticsearch_test.yaml b/deploy/eck-elasticsearch/templates/tests/elasticsearch_test.yaml
index c250bae142..7499e3f731 100644
--- a/deploy/eck-elasticsearch/templates/tests/elasticsearch_test.yaml
+++ b/deploy/eck-elasticsearch/templates/tests/elasticsearch_test.yaml
@@ -165,3 +165,17 @@ tests:
       - equal:
           path: spec.nodeSets[3].name
           value: cold
+  - it: should render es user and roles properly
+    set:
+      auth:
+        fileRealm:
+          - secretName: filerealm-secret-1
+        roles:
+          - secretName: roles-secret-1
+    asserts:
+      - equal:
+          path: spec.auth.fileRealm[0].secretName
+          value: filerealm-secret-1
+      - equal:
+          path: spec.auth.roles[0].secretName
+          value: roles-secret-1
```

File: deploy/eck-elasticsearch/values.yaml (hunk 1 above)
Comment (oziey88) [reply deploy/eck-elasticsearch/values.yaml]: Thank you for the review. 

unit test has been added, also updated the description.

**Review by thbkrkr — APPROVED** [review]
LGTM

**Comment by thbkrkr** [comment]
Jenkins test this please

**Comment by thbkrkr** [comment]
buildkite test this please

**Pull request merged**